integration-tests/tests/src/tests/observable.ts

```diff
@@ -404,25 +435,120 @@ describe("Observable", () => {
         },
       });
     });
+
+    describe("key-path filtered", () => {
+      it("calls listener only on relevant changes", async function (this: RealmObjectContext<Person>) {
+        await expectObjectNotifications(
+          this.object,
+          [
+            EMPTY_OBJECT_CHANGESET,
+            () => {
+              // Updating the age to 42 will ensure the object doesn't leave the results
+              this.realm.write(() => {
+                this.object.name = "Alex";
+                this.object.age = 42;
+              });
+            },
+            {
+              deleted: false,
+              changedProperties: ["name"],
+            },
+            () => {
```

Comment:
I wonder if there's a way we could make it clearer in the utils that this action will not trigger?  Perhaps a sort of `no-op` result given.

Reply:
Having to explicitly pass that, I fear we'll instead forget to pass it 🤔

Reply:
Then documentation would suffice.

Reply:
I updated the documentation of a shared `expectNotifications` utility function now.